src/Raven.Client/Documents/Operations/Operation.cs

```diff
@@ -115,15 +116,19 @@ protected virtual async Task Process()
                         var observable = changes.ForOperationId(_id);
                         _subscription = observable.Subscribe(this);
                         await observable.EnsureSubscribedNow().ConfigureAwait(false);
-                        
+
+                        if (_requestExecutor.ForTestingPurposes?.WaitBeforeFetchOperationStatus != null)
+                            _requestExecutor.ForTestingPurposes.WaitBeforeFetchOperationStatus.Wait();
+
                         // We start the operation before we subscribe,
                         // so if we subscribe after the operation was already completed we will miss the notification for it. 
                         await FetchOperationStatus().ConfigureAwait(false);
                     }
-                    catch
+                    catch(DatabaseDoesNotExistException)
```

Comment:
Wondering if we should only catch `DatabaseDoesNotExistException` like this or instead take `FetchOperationStatus()` (might throw) out of the try clause, so we don't dispose the connection on a "did not get reply for operation".
`DatabaseDoesNotExistException` is supposedly the only thing that should be thrown and not ignored in `DoWork`

Reply:
what if we really timeout?